spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/intermediate/IRconvertSpec.scala

```diff
@@ -91,6 +92,144 @@ class IRconvertSpec extends BigDLSpecHelper {
     Graph(conv1, output)
   }
 
+  def keras(classNum: Int, shape: Shape = Shape(28, 28, 3)): nn.keras.Sequential[Float] = {
+    import com.intel.analytics.bigdl.nn.keras._
+    import com.intel.analytics.bigdl.utils.Shape
+
+    val model = Sequential()
+    model.add(Convolution2D(6, 5, 5, activation = "tanh",
+      dimOrdering = "tf", inputShape = shape).setName("conv1_5x5"))
+    model.add(BatchNormalization(dimOrdering = "tf")).setName("bnbn")
+    model.add(MaxPooling2D(dimOrdering = "tf"))
+    model.add(Convolution2D(12, 5, 5, activation = "tanh", dimOrdering = "tf")
+      .setName("conv2_5x5"))
+    model.add(BatchNormalization(dimOrdering = "tf")).setName("bnbn22")
+    model.add(MaxPooling2D(dimOrdering = "tf"))
+    model.add(Flatten())
+    model.add(Dense(100, activation = "tanh").setName("fc1"))
+    model.add(Dense(classNum, activation = "softmax").setName("fc2"))
+    model
+  }
+
+  "Convert Blas with NHWC" should "be correct" in {
+    System.setProperty("bigdl.engineType", "mkldnn")
+    val input = Tensor[Float](2, 28, 28, 1).rand()
+    val gradOutput = Tensor[Float](2, 10).rand()
+
+    val blas = modelBlas(format = DataFormat("NHWC")).asInstanceOf[StaticGraph[Float]]
+    blas.setInputFormats(Seq(Memory.Format.nhwc))
+    blas.setOutputFormats(Seq(Memory.Format.nc))
+    val dnn = blas.cloneModule().toIRgraph()
+
+    val outBlas = blas.forward(input)
+    val outDnn = dnn.forward(input)
+    Equivalent.nearequals(outDnn.toTensor, outBlas.toTensor, 1e-4) should be (true)
+
+    val gradInputBlas = blas.backward(input, gradOutput)
+    val gradInputDnn = dnn.backward(input, gradOutput).toTensor[Float]
+    Equivalent.nearequals(gradInputDnn.toTensor, gradInputBlas.toTensor, 1e-4) should be (true)
+    System.clearProperty("bigdl.engineType")
+  }
+
+  // todo: use those unit tests after supporting keras conversion
+
+//  "Running keras with mkldnn" should "be correct"in {
+//    System.setProperty("bigdl.engineType", "mkldnn")
+//    val model = keras(classNum = 10)
+//
+//    val blas = model.toGraph().asInstanceOf[StaticGraph[Float]]
+//    blas.setInputFormats(Seq(Memory.Format.nhwc))
+//    blas.setOutputFormats(Seq(Memory.Format.nc))
+//
+//    val dnn = blas.cloneModule().asInstanceOf[StaticGraph[Float]].toIRgraph()
+//
+//    val input = Tensor[Float](2, 28, 28, 3).rand()
+//
+//    val out1 = blas.forward(input)
+//    val out2 = dnn.forward(input)
+//    Equivalent.nearequals(out1.toTensor[Float], out2.toTensor[Float], 1e-5) should be (true)
+//
+//    val gradOutput = Tensor[Float]().resizeAs(out1.toTensor[Float])
+//
+//    val gradInputBlas = blas.backward(input, gradOutput)
+//    val gradInputDnn = dnn.backward(input, gradOutput)
+//    Equivalent.nearequals(gradInputDnn.toTensor, gradInputBlas.toTensor, 1e-4) should be (true)
+//    System.clearProperty("bigdl.engineType")
```

Comment:
If it's not used, please delete the comments.

Reply:
those test codes will be opened after mengceng's code merged.